vmdb/app/models/ems_openstack_infra.rb

```diff
@@ -1,8 +1,24 @@
 class EmsOpenstackInfra < EmsInfra
   include EmsOpenstackMixin
 
+  before_create :ensure_parent_provider
+
   has_many :orchestration_stacks, :foreign_key => :ems_id, :dependent => :destroy
 
+  def ensure_parent_provider
+    # TODO(lsmola) this might move to a general management of Providers, but for now, we will ensure, every
+    # EmsOpenstackInfra has associated a Provider. This relation will serve for relating EmsOpenstackInfra
+    # to possible many EmsOpenstacks deployed through EmsOpenstackInfra
+
+    # Name of ems is unique, we will expect provider with the same name
+    found_provider = ProviderOpenstackInfra.find_by_name(name)
```

Comment:
I'm thinking the Provider will just be ProviderOpenstack, not ProviderOpenstackInfra, since it's the bridge...then the ProviderOpenstack will have one EmsOpenstackInfra and many EmsOpenstack (Cloud)